nixos/modules/services/monitoring/datadog-agent.nix

```diff
@@ -231,7 +250,7 @@ in {
       datadog-agent = makeService {
         description = "Datadog agent monitor";
         preStart = ''
-          chown -R datadog: /etc/datadog-agent
+          chown -R ${cfg.permissions.User}: /etc/datadog-agent
```

Comment:
Does datadog need to write to this directory? If not, you could replace this line with `serviceConfig.ConfigurationDirectory = "datadog-agent"` and drop the ([deprecated](https://github.com/NixOS/nixpkgs/issues/53852)) `serviceConfig.PermissionsStartOnly`